test/test_gpu3.cpp

```diff
@@ -8526,6 +8655,7 @@ TEST_F(NVFuserTest, FusionLayerNormFusedOpsRedundantCast_CUDA) {
   auto cg_outputs = fec.runFusionWithInputs(inputs);
```

Comment:
can you also add a check to make sure the reduandant casts are removed for this case? similar to `    TORCH_CHECK(
        cast_op_count == 0, "cast optimization isn't working as expected");`